src/ecs/wizards/executeCommand.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ecsExecToolkitGuideUrl } from '../../shared/constants'
+import { createCommonButtons } from '../../shared/ui/buttons'
+import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { WIZARD_EXIT, Wizard, WIZARD_BACK } from '../../shared/wizards/wizard'
+import { EcsContainerNode } from '../explorer/ecsContainerNode'
+
+export interface CommandWizardState {
+    task: string
+    command: string
+    confirmation?: 'yes' | 'supress'
+}
+
+function createTaskPrompter(node: EcsContainerNode) {
+    const taskItems = (async function () {
+        const taskArns = await node.listTasks()
+        if (taskArns.length === 0) {
+            return []
+        }
+        // Filter for only 'Running' tasks
+        const runningTasks = (await node.describeTasks(taskArns)).filter(t => {
+            return t.lastStatus === 'RUNNING' && t.desiredStatus === 'RUNNING'
+        })
+        return runningTasks.map(task => {
+            // The last 32 digits of the task arn is the task identifier
+            return {
+                label: task.taskArn!.substr(-32),
+                detail: `Status: ${task.lastStatus}  Desired status: ${task.desiredStatus}`,
+                data: task.taskArn!.substr(-32),
```

Comment:
if `data` allows undefined, I would suggest `?` instead of `!`. If it doesn't allow undefined, coalesce to a defaultvalue such as empty string or a fixed string like "unknown".

Same for `label`.

```suggestion
                data: task.taskArn?.substr(-32) ?? 'unknown',
```

Reply:
Even better would be to check for undefined closer to the source so these checks don't leak through the code.